src/integration-test/java/com/mongodb/jdbc/integration/AuthX509IntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.jdbc.integration;
+
+import static com.mongodb.jdbc.MongoConnection.X509_CLIENT_CERT_PATH;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.FileNotFoundException;
+import java.sql.*;
+import java.util.Properties;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class AuthX509IntegrationTest {
+
+    private static final String LOCAL_PORT_ENV_VAR = "LOCAL_MDB_PORT_ENT";
+    private static final String PASSWORD_ENV_VAR = "ADF_TEST_LOCAL_PWD";
+    private static final String X509_CERT_PATH_PROPERTY = "x509PemPath";
```

Comment:
We're seeing a massive proliferation of environmental variables that are dependent on evergreen or setting them locally. I'd like to start seeing accompanying shell files that encapsulate all of them that could be sourced both locally and in evergreen to make things more portable.

Reply:
I think for this one I'll create a ticket and we can think of how to improve the environment variable experience.